deploy.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+set -e # Exit with nonzero exit code if anything fails
```

Comment:
I'm pretty sure you got this (and the exit 0 if its not a deploy) from the script, but do we instead want to fail if the docs don't get generated? That would be annoying when pushing out code but would make sure we don't have failing docs, and if we set it up as a part of Travis should be easy to catch.

Reply:
The ```jsdoc``` command fails if it encounters invalid documentation, so the ```set -e``` catches it and should cause Travis to fail.

Reply:
I wrote a hacky workaround to determine whether the doc generation fails and save the error log, but I think the ```set -e``` is better for two reasons:
1. The doc generation outputs the error logs to stderr, so we can look at them on Travis
2. It will catch any other failures that might occur in the script

Reply:
Thats fine, I didn't look at the `set -e` command and just assumed it was for ignoring errors. If its catching them and lifting them up to Travis then it's doing exactly what I would want it to do.